### DISCUSSION TITLE
Adding color palettes to charts and dashboards

### DISCUSSION THREAD

**Discussion by ziggekatten** [apache/superset#21335] [General]
I'm trying to get my head around adding our own palettes, and as I understand it there are two entities that can be used in config.py:
EXTRA_SEQUENTIAL_COLOR_SCHEMES  and EXTRA_CATEGORICAL_COLOR_SCHEMES.

Question number one: What are the difference between the two? There is very little (to be diplomatic) documentation around this.

For fun, I spun up the docker instance on my devstation, modified my superset_config.py with
EXTRA_CATEGORICAL_COLOR_SCHEMES = [
    {
        "id": 'myCustomColors',
        "description": 'My custom Color',
        "label": 'My custom Color',
        "colors": ['#6BD3B3', '#6BD3B3', '#6BD3B3', '#66CBE2']
    }]

and 
EXTRA_SEQUENTIAL_COLOR_SCHEMES = [{
        "id": 'myCustomColors',
        "description": 'Test',
        "isDiverging": True,
        "label": 'Test ',
        "colors": ['#6BD111', '#6BD3B3', '#66CBE2']
    }]

Sure enough, in my chart I now get the following:
![image](https://user-images.githubusercontent.com/55629049/188550692-3c403013-2a2d-4e69-9d5c-7e7251059ce4.png)

So it seems that EXTRA_CATEGORICAL_COLOR_SCHEMES bites. Now to my issues...

I tried to rename description, label and colors, but changes simply does not bite. Is the info stored somewhere else than in the config.py?

Are the two EXTRA_CATEGORICAL_COLOR_SCHEMES and EXTRA_SEQUENTIAL_COLOR_SCHEMES related or dependent on each other?

Anyone that knows how this works?

**Comment by sfirke** [comment]
Your implementation of `EXTRA_CATEGORICAL_COLOR_SCHEMES` looks good.  In my config file I also have `"isDefault": True`.  When I've changed the colors in the config file and then done `docker-compose down` and then `up` I've seen the changes reflected right away, that info lives only in the config file(s).  I say file(s) plural because I put this code block in `superset_config_docker.py` but it could also go in `config.py` as you've done.  Easiest to maintain if it's all in one place.

**Reply by ziggekatten** [reply]
Embarrasingly, I have succeeded to put stuff in superset_docker_config.py that of course did an override on my   superset_config. Stupid, stupid, stupid.....It works as it should now.

Still, the question remains: What is the difference between EXTRA_SEQUENTIAL_COLOR_SCHEMES and EXTRA_CATEGORICAL_COLOR_SCHEMES?

**Reply by SkaarFacee** [reply]
If you have found the difference I would like to know as well

**Reply by syafiqmuda** [reply]
have you found the difference?

**Comment by RakshaDevadiga** [comment]
Hi all,

I faced some issue when I do like this. I have modify the superset_config.py file like EXTRA_CATEGORICAL_COLOR_SCHEMES = [
{
"id": 'myCustomColors',
"description": 'My custom Color',
"label": 'My custom Color',
"colors": ['#6BD3B3', '#6BD3B3', '#6BD3B3', '#66CBE2']
}] in order to add a new color pallet to my dashboard. But after doing changes here when I run docker compose up and browse localhiost:8088 where I created my dashboards I cant able to see new color palate while editing the dashboard properties.

I even tried docker compose build before browsing localhost:8088.

can anybody help me out to find the solution?

**Reply by simon-b06** [reply]
You need to add the following "FEATURE_FLAGS" -> "EXTRA_CATEGORICAL_COLOR_SCHEMES": True

Let me know if it works.
